core/tools/callTool.ts

```diff
@@ -141,43 +142,46 @@ export async function callTool(
   contextItems: ContextItem[];
   errorMessage: string | undefined;
 }> {
-  const args = JSON.parse(callArgs || "{}");
-  const uri = tool.uri ?? tool.function.name;
   try {
+    const args = JSON.parse(callArgs || "{}");
     let contextItems: ContextItem[] = [];
-    switch (uri) {
-      case BuiltInToolNames.ReadFile:
-        contextItems = await readFileImpl(args, extras);
-        break;
-      case BuiltInToolNames.CreateNewFile:
-        contextItems = await createNewFileImpl(args, extras);
-        break;
-      case BuiltInToolNames.GrepSearch:
-        contextItems = await grepSearchImpl(args, extras);
-        break;
-      case BuiltInToolNames.FileGlobSearch:
-        contextItems = await fileGlobSearchImpl(args, extras);
-        break;
-      case BuiltInToolNames.RunTerminalCommand:
-        contextItems = await runTerminalCommandImpl(args, extras);
-        break;
-      case BuiltInToolNames.SearchWeb:
-        contextItems = await searchWebImpl(args, extras);
-        break;
-      case BuiltInToolNames.ViewDiff:
-        contextItems = await viewDiffImpl(args, extras);
-        break;
-      case BuiltInToolNames.LSTool:
-        contextItems = await lsToolImpl(args, extras);
-        break;
-      case BuiltInToolNames.ReadCurrentlyOpenFile:
-        contextItems = await readCurrentlyOpenFileImpl(args, extras);
-        break;
-      case BuiltInToolNames.CreateRuleBlock:
-        contextItems = await createRuleBlockImpl(args, extras);
-        break;
-      default:
-        contextItems = await callToolFromUri(uri, args, extras);
+    if (tool.uri) {
+      contextItems = await callToolFromUri(tool.uri, args, extras);
```

Comment:
```suggestion
     const contextItems = await getContextItems(...)
```

I have it on my to-do list to eventually lint out all the instances of `let` to make our code more predictable.